google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/OpenTelemetryPubsubTracer.java

```diff
@@ -0,0 +1,460 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.pubsub.v1;
+
+import com.google.pubsub.v1.PubsubMessage;
+import com.google.pubsub.v1.SubscriptionName;
+import com.google.pubsub.v1.TopicName;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanBuilder;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.api.trace.StatusCode;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import java.util.List;
+
+public class OpenTelemetryPubsubTracer {
+  private final Tracer tracer;
+  private boolean enabled = false;
+
+  private static final String PUBLISH_FLOW_CONTROL_SPAN_NAME = "publisher flow control";
+  private static final String PUBLISH_BATCHING_SPAN_NAME = "publisher batching";
+  private static final String SUBSCRIBE_CONCURRENCY_CONTROL_SPAN_NAME =
+      "subscriber concurrency control";
+  private static final String SUBSCRIBE_SCHEDULER_SPAN_NAME = "subscriber scheduler";
+
+  private static final String MESSAGE_SIZE_ATTR_KEY = "messaging.message.body.size";
```

Comment:
Some of these attributes (most things that start with messaging.message) are defined in semantic conventions, but I notice that it is marked as deprecated [here](https://github.com/open-telemetry/semantic-conventions-java/blob/dd1d40c6be372bc73e203307d4a02c24c5ce89f2/semconv/src/main/java/io/opentelemetry/semconv/SemanticAttributes.java#L819)

I'm ok with hard coding for now if there isn't a suitable replacement. Is that accurate?

Reply:
Used MessagingIncubatingAttributes for this attribute